library/src/main/java/com/novoda/downloadmanager/LiteDownloadBatchStatus.java

```diff
@@ -41,10 +41,6 @@ public void update(long currentBytesDownloaded, long totalBatchSizeBytes) {
         this.bytesDownloaded = currentBytesDownloaded;
         this.totalBatchSizeBytes = totalBatchSizeBytes;
         this.percentageDownloaded = getPercentageFrom(bytesDownloaded, totalBatchSizeBytes);
-
-        if (this.bytesDownloaded == this.totalBatchSizeBytes && this.totalBatchSizeBytes != ZERO_BYTES) {
-            this.status = Status.DOWNLOADED;
```

Comment:
Only updating in memory, never hits the persistence layer 😬

Reply:
Are you 100% sure of this? the update is called from `DownloadsBatchPersistence.loadAsync()`

Reply:
100% sure, there is a check for `DOWNLOADED` above that instantly returns so if it was ever modified `in-memory` then it would never hit the persistence. I tested this quite a lot manually and with the debugger when I realised there was a problem.

Reply:
Yeah I know but the load async is different, because there the file status is set to downloaded immediately, doesn't need to be queued to be downloaded, processed and then is immediately changed to downloaded. But if you have tested that the loadAsync works fine for downloaded files, all good